runtime/src/main/java/io/novafoundation/nova/runtime/multiNetwork/connection/ChainConnection.kt

```diff
@@ -88,16 +86,21 @@ class ChainConnection internal constructor(
     ).shareIn(scope = this, started = SharingStarted.Eagerly)
 
     private val chain = MutableStateFlow(initialChain)
+
     private val availableNodes = chain.map { it.nodes }
-        .shareIn(scope = this, started = SharingStarted.Eagerly, replay = 1)
+        .distinctUntilChanged()
+        .share(SharingStarted.Eagerly)
 
-    val currentUrl = chain.flatMapLatest { getNodeUrlFlow(it) }
-        .shareIn(scope = this, started = SharingStarted.Eagerly, replay = 1)
+    val currentUrl = nodeAutobalancer.connectionUrlFlow(
```

Comment:
We dont need to flatMap on chain since the `connectionUrlFlow` only uses chainId